lib/tapioca/gem/listeners/mixins.rb

```diff
@@ -62,6 +65,23 @@ def add_mixins(tree, mods, mixin_type)
             end
         end
 
+        sig do
+          params(
+            constant: Module,
+            mixin: Module,
+            mixin_type: Runtime::Trackers::Mixin::Type
+          ).returns(T::Boolean)
+        end
+        def mixed_in_by_gem?(constant, mixin, mixin_type)
+          mixin_location =
+            T.cast(
+              Runtime::Trackers::Mixin.constants_with_mixin(mixin).dig(mixin_type, constant),
+              T.nilable(String)
+            )
+
+          !!mixin_location && @pipeline.gem.contains_path?(mixin_location)
+        end
```

Comment:
@Morriar I took your advice here and combined identifying the mixin location and determining whether it is in the gem path into one method. I did some stuff to get type checking to work, like casting the mixin location and then adding `!!` to have it always return a boolean. Let me know if there is a simpler way to resolve these issues?